compiler-rt: fix toolchain installer

leon-anavi opened the pull request:
Fix building a toolchain installer with bitbake
-c populate_sdk in packages compiler-rt,
compiler-rt-dev and compiler-rt-staticdev.

Signed-off-by: Leon Anavi <leon.anavi@konsulko.com>

derekstraka commented:
Once validated, these need to go into master and krogoth branches

jmesmon commented:
Shouldn't we instead:

 - allow compiler-rt to be empty
 - keep the *.a files in compiler-rt-dev

I don't believe the *.a is used at runtime, only build time.

derekstraka commented:
^^ That was going to be my comment as well

leon-anavi commented:
OK, I will try to improve the patch.

leon-anavi commented:
I have updated the patch. Now libcompiler-rt.a is in compiler-rt-devstatic. The rest of the changes are adjustments to make sure that both: bitbake core-image-minimal and bitbake core-image-minimal -c populate_sdk work.

Best regards,
Leon

jmesmon commented:
```
ALLOW_EMPTY_${PN} = "1"
```
Should likely be used in place of including `${libdir}` in PN so it is generated. IIRC, would allow the empty compiler-rt package to be generated.

cardoe commented:
Currently failing to build `libstd-rs` with the following error during linking:
```
|   = note: /var/build/build/tmp/sysroots/x86_64-linux/usr/libexec/x86_64-poky-linux/gcc/x86_64-poky-linux/5.3.0/ld: cannot find -lcompiler-rt
```

leon-anavi commented:
Hi @cardoe, @jmesmon, @derekstraka,

Thank you for the support and the guidelines. I have updated the patch. This time I used as an example the information from the Yocto mailing list how to include static library and headers in sdk and a another patch for fortran. 

As @jmesmon recommended I have added ALLOW_EMPTY_${PN} = "1". I have added RDEPENDS_${PN}-dev = "" so compiler-rt-dev and compiler-rt-staticdev do not depend on the binary package. The *.a files are in compiler-rt-devstatic an -dev is empty. 

Thanks,
Leon

cardoe commented on recipes-devtools/rust/compiler-rt_1.10.0.bb:
```diff
@@ -26,5 +32,12 @@ do_compile () {
 	mkdir -p ${D}${libdir}
+}
+
+do_install () {
```
Why is `do_install()` duplicated? I think this was probably meant to be squished together?

cardoe commented on recipes-devtools/rust/compiler-rt_1.12.1.bb:
```diff
@@ -22,3 +26,6 @@ do_install () {
 }
+
+FILES_${PN}-dev = ""
+FILES_${PN}-staticdev = "${libdir} ${libdir}/*.a"
```
Shouldn't this just be:
```
FILES_${PN}-staticdev = "${libdir}/*.a"
```

cardoe commented on recipes-devtools/rust/compiler-rt_1.10.0.bb:
```diff
@@ -26,5 +32,12 @@ do_compile () {
 }
+
+FILES_${PN}-dev = ""
+FILES_${PN}-staticdev = "${libdir} ${libdir}/*.a"
```
Shouldn't this just be:
```
FILES_${PN}-staticdev = "${libdir}/*.a"
```

leon-anavi commented:
@cardoe thank you for the code review. Following your recommendations I have improved the patch. Sorry for the duplicated do_install. I am a bit distrait with all the noise around the holidays :)

Best regards,
Leon

cardoe commented:
I tested this locally in krogoth and master. It passed in krogoth but master fails with populate_sdk due to python3-native. I'll forward port this patch into morty and master.
